polytracker/taint_dag.py

```diff
@@ -314,6 +338,7 @@ def __init__(self, file: BinaryIO) -> None:
         self.sink_cache: Dict[int, TDSink] = {}
 
         self.fd_headers: List[Tuple[Path, TDFDHeader]] = list(self.read_fd_headers())
+        self.fn_headers: List[Tuple[str, TDFnHeader]] = list(self.read_fn_headers())
```

Comment:
I guess this could produce some overhead on large traces (with many functions). But, maybe it isn't that many. We could always change it to be lazy at some later point if needed. The reasoning for `fd_headers` is probably that there aren't that many files being handled in a TDAG file.
Anyhow, for now I think it is good that they follow the same pattern.

Reply:
Agreed.